packages/rocketchat-message-snippet/server/requests.js

```diff
@@ -0,0 +1,47 @@
+/* global Cookies */
+WebApp.connectHandlers.use('/snippet/download', function(req, res) {
+	var match = /^\/([^\/]+)\/(.*)/.exec(req.url);
+
+	if (match[1]) {
+		let snippet = RocketChat.models.Messages.findOneById(match[1]);
```

Comment:
And, can you verify if the user has access to the room where this message came from?